virtual-desktop/src/app/application-manager/application-manager.service.ts

```diff
@@ -217,17 +237,60 @@ export class ApplicationManager implements MVDHosting.ApplicationManagerInterfac
 
   showApplicationWindow(plugin: DesktopPluginDefinitionImpl): void {
     const windowManager: MVDWindowManagement.WindowManagerServiceInterface = this.injector.get(MVDWindowManagement.Tokens.WindowManagerToken);
-    const windowId = windowManager.getWindow(plugin);
+    const windowId = this.runningPluginAppMap.get(plugin.getIdentifier());
     if (windowId != null) {
-      windowManager.showWindow(windowId);
+      windowManager.showWindow(windowId[0]);
     } else {
       this.spawnApplication(plugin, null);
     }
   }
-
-  isApplicationRunning(plugin: DesktopPluginDefinitionImpl): boolean {
+ 
+  getAppPropertyInformation(plugin: DesktopPluginDefinitionImpl):any{
+    const pluginImpl:DesktopPluginDefinitionImpl = plugin as DesktopPluginDefinitionImpl;
+    const basePlugin = pluginImpl.getBasePlugin();
+    return {"isPropertyWindow":true,
+    "appName":pluginImpl.defaultWindowTitle,
+    "appVersion":basePlugin.getVersion(),
+    "appType":basePlugin.getType(),
+    "copyright":pluginImpl.getCopyright(),
+    "image":plugin.image
+    };    
+  }
+  
+  showApplicationPropertiesWindow(plugin: DesktopPluginDefinitionImpl): void {
     const windowManager: MVDWindowManagement.WindowManagerServiceInterface = this.injector.get(MVDWindowManagement.Tokens.WindowManagerToken);
-    return windowManager.getWindow(plugin) != null;
+    const pluginID = plugin.getIdentifier();
+    let windowId = this.pluginPropertyShowingMap.get(pluginID);
+    if (windowId==undefined){
+      this.pluginManager.findPluginDefinition("org.zowe.zlux.appmanager.app.propview").then(viewerPlugin => {
```

Comment:
The application manager shouldn't be hardcoding this. The application manager should only spawn an application when given an application ID, via spawnapplication. See my longer comment in https://github.com/zowe/zlux-platform/pull/7